src/Components/Confirm/Confirm.js

```diff
@@ -0,0 +1,215 @@
+   */
+  primary: PropTypes.bool,
+  /**
+   * Indicate a destructive action
+   */
+  danger: PropTypes.bool,
+  /**
+   * Render the button as inline text without padding
+   */
+  plain: PropTypes.bool,
+  /**
+   * optional title of the modal. If set, a header will be added to the dialog
+   */
+  title: PropTypes.string,
+  /**
+   * Callback after the user has successfully confirmed their intent.
+   */
+  onConfirm: PropTypes.func.isRequired,
+  /**
+   * Callback after the user does NOT confirm their intent
+   */
+  onCancel: PropTypes.func,
+  /**
+   * Contents of the confirm button in the modal
+   */
+  confirmButtonContent: PropTypes.node,
+  /**
+   * Contents of the cancel button in the modal
+   */
+  cancelButtonContent: PropTypes.node,
+  /**
+   * If provide the user must type this before they can confirm.
+   */
+  confirmValue: PropTypes.string,
+  /**
+   * Elements to be rendered as children of the modal
+   */
+  confirmationContent: PropTypes.node,
+  /**
+   * When comparing what a user types to the value should case be ignored.
+   */
+  ignoreCase: PropTypes.bool,
+};
+
+const defaultProps = {
+  title: 'Are you sure?',
+  confirmButtonContent: 'Confirm',
+  cancelButtonContent: 'Cancel',
+  ignoreCase: false,
+};
+
+function Confirm({
+  disabled,
+  primary,
+  plain,
+  danger,
+  children,
+  className,
+  onConfirm,
+  onCancel,
+  icon,
+  iconAfterText,
+  title,
+  confirmButtonContent,
+  cancelButtonContent,
+  confirmValue,
+  confirmationContent,
+  ignoreCase,
+  ...rest
+}) {
+  const requiresTyping = confirmValue !== undefined;
+
+  const [showModal, setShowModal] = useState(false);
+  const [confirmed, setConfirmed] = useState(!requiresTyping);
+  const [inputValue, setInputValue] = useState('');
+
+  if (!confirmationContent) {
+    if (requiresTyping) {
+      confirmationContent = (
+        <p>
+          Please confirm you are sure by entering{' '}
+          <Text appearance="code">{confirmValue}</Text> into the input
+          below.
+        </p>
+      );
+    } else {
+      confirmationContent = <p>Are you sure you want to continue?</p>;
+    }
+  }
+
+  const internalOnConfirm = () => {
+    setShowModal(false);
+    setInputValue('');
+    if (onConfirm) {
+      onConfirm();
+    }
+  };
+
+  const internalOnCancel = () => {
+    setShowModal(false);
+    setInputValue('');
+    if (onCancel) {
+      onCancel();
+    }
+  };
+
+  const handleInputChange = e => {
+    setInputValue(e.target.value);
+  };
+
+  useEffect(() => {
+    if (requiresTyping) {
+      if (ignoreCase) {
+        setConfirmed(
+          inputValue.toLowerCase() === confirmValue.toLowerCase(),
+        );
+      } else {
+        setConfirmed(inputValue === confirmValue);
+      }
+    }
+  }, [inputValue, confirmValue, ignoreCase]);
+
+  return (
+    <>
+      <Button
+        primary={primary}
+        danger={danger}
+        plain={plain}
+        className={className}
+        disabled={disabled}
```

Comment:
does not need to be added and can be removed from the props as it is included in `...rest`. Same for primary, plain, and className

Reply:
This is actually a good time to talk about this. It does need to be added. based on how styleguideist generates the docs to pick up defaults and documentation you need it called out you can use the `...rest` param. I grabbed out the items that I thought would be common and I wanted to document. 

We could likely import some sort of shared types and make it work but for now this is how we have been working around this limitation. Also, given our discussions about moving to storybook for docs I'd rather tackle this then.

Reply:
Interesting that it looks at the actual component and not the propTypes.

Reply:
it looks at both  and combines them (at least from my testing)